### PR TITLE
Improve `repr` for `SequenceElem`

### PR DESCRIPTION
**Pull request by MDLC01**
This small PR changes the `repr` for sequences so that it it valid Typst code and less confusing.

For comparison, the old `repr` looked like the following:
```
[(
  [A],
  h(10pt),
  bold(body: [B]),
)]
```

And here is the new one:
```
(
  [A]
    + h(10pt)
    + bold(body: [B])
)
```

Another solution was to use the join syntax:
```
{
  [A]
  h(10pt)
  bold(body: [B])
}
```
but I think the `+` syntax is more explicit.

### CODE REVIEW

**Comment by Leedehai** [comment]
I have some reservations about the `+` symbol, because it adds "noise" that is not part of the sequence itself, and it may confuse people (especially if the elements contain `+` somewhere themselves..).

I think the original is fine, or the alternative one i.e. `{...}`.

**Comment by MDLC01** [comment]
> I think the original is fine

I have to disagree on this specific point. The reason why I made this PR in the first place is because I saw the current repr while debugging a Typst template, and I was unable to determine what this was representing. I had to look at different `Repr` implementations in the compiler to understand what it was, because it is so far from any existing syntax.

I'm fine with using the join syntax (`{..}`) instead, though. But I prefer to wait for at least one other feedback before (we have time, it's not like this needs to be included in 0.11.0).

**Comment by PgBiel** [comment]
The original is indeed a bit confusing, in the sense that it seems to try to mimic markup and code mode at the same time. Perhaps a better (and more accurate) representation is `sequence(a, b, c, ...)`.

**Comment by laurmaedje** [comment]
The current one is indeed pretty weird, don't know where that came from. I'm not a big fan of the one with addition. From the ones in the PR description, I like the code block one best, but `sequence(..)` is probably the clearest one. I also intend to make `sequence` public alongside the rework from element functions into types.

**Comment by laurmaedje** [comment]
There is a conflict now. There's also a problem with the checks apparently, but I think that would be fixed just by merging main into this PR.

**Comment by MDLC01** [comment]
I'm working on the conflict.

**Comment by MDLC01** [comment]
All good now!

**Comment by MDLC01** [comment]
All good now!

File: MDLC01/test.typ
Comment (laurmaedje) [inline MDLC01/test.typ:1]: some test files ended up here

**Comment by laurmaedje** [comment]
Thanks!

File: MDLC01/test.typ
Comment (MDLC01) [reply MDLC01/test.typ:1]: Oops, sorry. I swear I wasn't trying to add a backdoor 😬

**Pull request merged**